### PR TITLE
Add smoketest of cvr, manifest uploads to server

### PR DESCRIPTION
**Pull request by nealmcb**
At this point this is very messy, with some very very long lines of static data in auto-generated code,
no error handling in the main.py upload code, total kludge for dealing with ordering differences and ignoring timestamps (just do a fuzzy compare of the length of the response), etc.

But it runs a smoketest, testing several of the endpoints in corla-server-test.html,
and provides a basis for comment and refinement.

### CODE REVIEW

File: test/smoketest/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+The current tests are based on cvr at `test/e-1/arapahoe-regent-3-clear-CVR_Export.csv` and the manifest at `test/e-1/arapahoe-manifest.csv`.
+
+To run the smoketest, first cd to this directory.
+
+Start up a fresh server, or reset its database to be empty.
+
+Run `./main.py` to upload the cvr and manifest files.
+
+Run `pytest server_test.py` to test the
```
Comment (ranweiler) [inline test/smoketest/README.md]: This sentence is accidentally truncated.

File: test/smoketest/main.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python
```
Comment (ranweiler) [inline test/smoketest/main.py]: Are expecting Python 2 or 3?

File: test/smoketest/main.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    test_get_contest()
+    test_get_cvr_county_Arapahoe()
+    test_get_ballot_manifest_county_Arapahoe()
+    test_get_contest_id_2()
+    test_get_contest_county_Arapahoe()
+
+if __name__ == "__main__":
+    upload_cvr("../e-1/arapahoe-regent-3-clear-CVR_Export.csv")
+    upload_manifest("../e-1/arapahoe-manifest.csv")
+    # server_sequence()
```
Comment (ranweiler) [inline test/smoketest/main.py]: Is this meant to be commented out?

Comment (nealmcb) [inline test/smoketest/main.py]: Yes, for now. If I like zerotest and pytest, I assume I can figure out how to run the tests from main and update the README.md.

I've also resolved the other issues.
Both Python 2 and Python 3 work, but I'm mainly testing with Python 3.
I've made note of that.

**Comment by nealmcb** [comment]
OK, this is updated to work on master 54a95d4, and be better documented and easier to update.
Please check it out and merge if you think it's useful.
Still lots to do to make it cover all the ASM transitions etc. which I'll continue in parallel or later.

**Comment by dmzimmerman** [comment]
As far as I can see, the SQL query totals _all_ CVRs in the database, not just uploaded ones (or, more specifically, uploaded ones that are actually under audit - in the case of multiple uploads by a county to fix some sort of mistake, there will be many records that are not under audit, unless we purge them from the database... which we can, but currently don't, do). That might be fine if we're only running it once before any auditing happens and saving the results somewhere, though.

**Comment by nealmcb** [comment]
Good point, @dmzimmerman. I think they will want a newly uploaded cvr or manifest to override any previous uploads, and not be included in these calculations. Offhand, for simplicity, I'd want all entries for that county removed from the database before adding the new one, and perhaps save the old file on the filesystem or something.
And @sfsinger19103, I think that at upload time, they will want the tabulation to be just for the subset uploaded by that county.
I think we do want to show results from all contests, since all contests are under audit. Only a few are used in the RLA calculations to set the sample size and calculate risk levels.

**Comment by dmzimmerman** [comment]
I don’t think we ever want to remove an uploaded file from the database. I’m willing to be convinced that it makes sense to delete a county’s set of parsed CVRs when they upload new ones though, since they could always be re-parsed, and have the cast vote record table represent only the ones we’re actively working with. It occurs to me that uploading the file and parsing the CVRs is currently a very long lived operation (the database is locked for the entire duration of the file upload), and that may not make sense. @ranweiler, @kiniry, do you think it might make more sense to have an endpoint for uploading files where you specify the file type and hash (as one of the multipart/form-data parameters), get back a file ID of some sort, and subsequently tell the server “go ahead and make that my active file of that type”? (this same discussion applies to ballot manifest files too)

**Comment by dmzimmerman** [comment]
Also, at the very least, I’d think this query would want to restrict itself to the set of CVRs for which the type is UPLOADED, because if we have to pick up where we left off at some point, we’d be out of luck unless we persistently save the tabulation results somewhere (which is do-able, of course).

**Comment by ranweiler** [comment]
> do you think it might make more sense to have an endpoint for uploading files where you specify the file type and hash (as one of the multipart/form-data parameters), get back a file ID of some sort, and subsequently tell the server “go ahead and make that my active file of that type”?

@dmzimmerman, so do you mean decoupling the file upload from parsing? In other words, I submit `myCvrExport.csv` to `/file-upload` as form data with a `type` field of value `"cvrExport"`, then get an `id` back for that file. Then I would `POST` that `id` to another endpoint like `/parse-cvr`? Or do you mean something else?

**Comment by kiniry** [comment]
Yes, I believe that is what @dmzimmerman is saying, @ranweiler.

File: test/smoketest/README.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+This smoketest has been tested against master commit 54a95d4.
+
+For now, you'll need to install
+[zerotest](https://github.com/jjyr/zerotest)
+to run these tests, and/or generate new ones in the same way.
+
+This is mainly tested with python3, but has been seen to work
+on python2 also.
+
+`pip install zerotest`
```
Comment (kiniry) [inline test/smoketest/README.md]: ... or 'pip3 install zerotest` (or similar) if you have multiple Python installations

File: test/smoketest/README.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+ curl http://localhost:8888/contest/id/71
+ curl http://localhost:8888/cvr/county?3
+ curl http://localhost:8888/acvr/county?3
+ curl http://localhost:8888/acvr
+EndOfInput
+```
+
+Stop the `zerotest server` process, e.g. with Cntl-C.
+
+Interrim test of captured queries:
```
Comment (kiniry) [inline test/smoketest/README.md]: s/Interrim/Interim/

File: test/smoketest/main.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python
+"""Smoketest the RLA server
+TODO: get server_sequence working and displaying errors
```
Comment (kiniry) [inline test/smoketest/main.py]: File an issue.

File: test/smoketest/smoketest-update.bash [hunk 7]
```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+set -eux -o pipefail
+
+# Put these somewhere else before running via travis
+export TRAVIS_BUILD_DIR=`git rev-parse --show-toplevel`
```
Comment (ranweiler) [inline test/smoketest/smoketest-update.bash]: Is this meant to be run by Travis? If so, these variables should just be in the script's environment. I don't _think_ you should have to re-export them for the child scripts, either. Did that turn out to be necessary?

File: server/eclipse-project/pom.xml [hunk 8]
```diff
@@ -4,9 +4,12 @@
 	<modelVersion>4.0.0</modelVersion>
 	<groupId>us.freeandfair.production</groupId>
 	<artifactId>colorado_rla</artifactId>
-	<version>0.0.1</version>
+	<version>0.7.0-alpha</version>
```
Comment (ranweiler) [inline server/eclipse-project/pom.xml]: I don't think this PR is where we should be cutting releases.

File: test/smoketest/main.py (hunk 6 above)
Comment (nealmcb) [inline test/smoketest/main.py]: Filed: https://github.com/FreeAndFair/ColoradoRLA/issues/254

File: test/smoketest/smoketest-update.bash (hunk 7 above)
Comment (nealmcb) [inline test/smoketest/smoketest-update.bash]: I have not tried running this from Travis yet, but I intend to after figuring out what the python install pieces will look like etc.
For now, I'm just using the same environmental variables, and this was a quick hack to make them easy to set.

Comment (nealmcb) [inline test/smoketest/smoketest-update.bash]: Putting the TEST directory in travis.yml just seems like a good hint to folks reviewing the travis setup that there is stuff close to being useful there. If you like I'll take it out, but it seems useful to me as is.

File: server/eclipse-project/pom.xml (hunk 8 above)
Comment (nealmcb) [inline server/eclipse-project/pom.xml]: That's fine. Just trying to keep the conversation going, and I wanted to test the release naming/launching process.
But this didn't seem like a particularly BAD PR to finally admit that we're nearing release, and can automatically test a bunch of stuff....
So I'll leave it in if I can get away with it :)

Comment (dmzimmerman) [inline server/eclipse-project/pom.xml]: I agree with @ranweiler. Let's not do this here.

Comment (nealmcb) [inline server/eclipse-project/pom.xml]: Makes sense.  I left in the fix for the build warnings.

We've been in 0.7.0-alpha for a week now, but I'll do it in a separate PR, after this PR lands.

File: test/smoketest/smoketest-update.bash (hunk 7 above)
Comment (nealmcb) [inline test/smoketest/smoketest-update.bash]: Removed, per feedback, in f3e381c

**Pull request merged**